### ISSUE TITLE
Ability to search, select all, then "select all messages that match this search"

### ISSUE THREAD

**Issue by jas8522** [the-djmaze/snappymail#300]
A common use-case for webmail is to clean up server-side storage. This is most common when you have your mailbox set to forward to another account or you use POP retrieval and primarily use an external mail system to manage your email.

The excellent filtering/search capabilities in SnappyMail make finding specific messages like those that are larger or those older than a specific date super easy, which is great! However if you have 10+ pages of search results, it can take some time to go through them all and remove messages.

Gmail does this well with this process flow:

1. Complete a search
2. Use the select all button to select all visible messages
3. It indicates how many messages have been selected, then provides a link that says "Select all conversations that match this search." Click that option. The indicator for how many messages have been selected changes to "All conversations in this search are selected." with the option to "Clear selection"
5. You can then press the trash button to remove all the messages from the search results. You can also move the messages to a folder / archive them, though IMO this is less important.

A workaround would be to allow seeing more than 20 messages per page, but even then if your search results are massive, it doesn't work as effectively as the Gmail solution.

**Comment by the-djmaze** [comment]
It would be nice to have this addition, but it needs some work.
Just POSTing all message ID's to server could result in timeouts.
It should be a batch process to handle 20-50 per request.

Websockets would be nice, but that would be a complete different webmail app with lot more server configuration (not that easy for most).

**Comment by jas8522** [comment]
@the-djmaze I'm not super familiar with data streaming code. It looks like you could send the entirety of the message IDs with JS through to the server-side stream responder (which adjusts its own timeouts / removal of buffering to be more efficient), rather than writing batch processing code. Am I reading that right?

**Comment by the-djmaze** [comment]
You are right. The code works but still needs special configuration of Apache configuration.
When Apache is not configured the right way, it will not work.

That's also why i haven't changed the core yet to pre-fetch messages in stream mode instead of 5-10 requests.

**Comment by jas8522** [comment]
I can see how it might make sense to implement this feature with stream as a first test. That way it's not core functionality that would be affected should the end-user not have the necessary web server configurations in place.

Would then modify the install instructions to include the necessary apache and/or nginx config.

I'll see if I can free up some time to pick away at this :)

**Comment by apezio** [comment]
> A workaround would be to allow seeing more than 20 messages per page, but even then if your search results are massive, it doesn't work as effectively as the Gmail solution.

For reference for anyone else:
"Settings -> Messages on Page".  Change it to a large number (I use 500).  Do what you need to do and change the setting back.
